pkg/sql/codegen/optimize/codegen.go

```diff
@@ -15,21 +15,84 @@ package optimize
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
 	"sqlflow.org/sqlflow/pkg/ir"
 	pb "sqlflow.org/sqlflow/pkg/proto"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/attribute"
 	"strings"
 	"text/template"
 )
 
+func checkIsPositiveInteger(i interface{}, name string) error {
+	if i.(int) <= 0 {
+		return fmt.Errorf("%s should be positive integer", name)
+	}
+	return nil
+}
+
+var attributeDictionary = attribute.Dictionary{
+	"data.enable_slice": {attribute.Bool, false, "Whether to enable data slicing", nil},
+	"data.batch_size":   {attribute.Int, -1, "Batch size when training", nil},
+	"worker.num": {attribute.Int, 1, "Worker number", func(i interface{}) error {
+		return checkIsPositiveInteger(i, "worker.num")
+	}},
+	"worker.core": {attribute.Int, 8, "Worker core number", func(i interface{}) error {
+		return checkIsPositiveInteger(i, "worker.core")
+	}},
+	"worker.memory": {attribute.Int, 4096, "Worker memory", func(i interface{}) error {
+		return checkIsPositiveInteger(i, "worker.memory")
+	}},
+	"solver.*": {attribute.Unknown, nil, "Solver options", nil},
+}
+
+const (
+	dataAttrPrefix   = "data."
+	solverAttrPrefix = "solver."
+	workerAttrPrefix = "worker."
```

Comment:
Either make sure that the construction of the `attribute.Dictionary` uses these prefixes too, or, if they are only used in `func GenerateOptflowOptimzieCode`, move them into that function definition.

Reply:
Done.